app/policies/application_policy.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class ApplicationPolicy
+  RESOURCE_ACTIONS = %i[index? create? new? update? edit? destroy?].freeze
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+    require_user! if user_required?
+  end
+
+  def method_missing(_method_name, *_args)
+    false
+  end
+
+  def respond_to_missing?(method_name, include_private = false)
+    RESOURCE_ACTIONS.include?(method_name) || super
+  end
+
+  private
+
+  def user_required?
+    true
+  end
+
+  def require_user!
+    raise Pundit::NotAuthorizedError unless @user
```

Comment:
If this exception is thrown, the error message is somewhat incomplete:

```
Pundit::NotAuthorizedError: not allowed to  this NilClass
```

This is due to the default message:

```rb
        message = options.fetch(:message) { "not allowed to #{query} this #{record.class}" }
```

Therefore, we could overwrite the message here it make it more accurate for fellow devs:

```rb
    raise Pundit::NotAuthorizedError.new("my custom error message") unless @user
```

Reply:
Good idea. I changed that.